src/main/java/com/atlassian/bitbucket/jenkins/internal/http/HttpRequestExecutorImpl.java

```diff
@@ -113,38 +155,27 @@ private void handleError(int responseCode, @Nullable String body, Headers header
     }
 
     private <T> T performRequest(Request request, ResponseConsumer<T> consumer) {
+        Response response = null;
         try {
-            Response response = httpCallFactory.newCall(request).execute();
-            int responseCode = response.code();
-
-            try (ResponseBody body = response.body()) {
-                if (response.isSuccessful()) {
-                    log.fine("Bitbucket - call successful");
-                    return consumer.consume(response);
-                }
-                handleError(responseCode, body == null ? null : body.string(), response.headers());
-            }
-        } catch (ConnectException | SocketTimeoutException e) {
-            log.log(Level.FINE, "Bitbucket - Connection failed", e);
-            throw new ConnectionFailureException(e);
-        } catch (IOException e) {
-            log.log(Level.FINE, "Bitbucket - io exception", e);
-            throw new BitbucketClientException(e);
-        } catch (RateLimitedException e) {
-            RetryOnRateLimitConfig rateLimitConfig = request.tag(RetryOnRateLimitConfig.class);
-            if (rateLimitConfig != null) {
-                if (rateLimitConfig.incrementAndGetAttempts() <= rateLimitConfig.getMaxAttempts()) {
-                    try {
-                        Thread.sleep(e.getRetryIn());
-                    } catch (InterruptedException ex) {
-                        throw new UnhandledErrorException("Interrupted during wait to retry", -2, null);
-                    }
-                    return performRequest(request, consumer);
-                }
+            response = handleRequest(request);
+        } finally {
+            if (response != null && response.body() != null) {
+                response.close();
             }
-            throw e;
         }
-        throw new UnhandledErrorException("Unhandled error", -1, null);
+        return consumer.consume(response);
+    }
+
+    private InputStream performStreamingGet(HttpUrl url, RequestConfiguration... additionalConfigs) {
+        Set<RequestConfiguration> additionalConfig = toSet(additionalConfigs);
+        Request.Builder requestBuilder = new Request.Builder().url(url);
+        additionalConfig.forEach(config -> config.apply(requestBuilder));
+        return performStreamingRequest(requestBuilder.build());
+    }
+
+    private InputStream performStreamingRequest(Request request) {
```

Comment:
can I propose we just inline this method? It is called just once from the method above and it is tiny. We can extract things out when we need it